src-java/kilda-model/src/main/java/org/openkilda/model/cookie/FlowSegmentCookie.java

```diff
@@ -133,7 +136,11 @@ private static long makeValue(CookieType type, FlowPathDirection direction, long
         if (direction != null) {
             value = makeValueDirection(direction);
         }
-        return setField(value, FLOW_EFFECTIVE_ID_FIELD, flowEffectiveId);
+        long result = setField(value, FLOW_EFFECTIVE_ID_FIELD, flowEffectiveId);
+        if (looped) {
```

Comment:
You loose `looped` flag in `.toBuilder().build()` cycle. Also you need getter for it.

Reply:
added